src/messages.rs

```diff
@@ -347,6 +337,8 @@ impl SignedMessage {
         for invalid_signature in &invalid_signatures {
             let _ = self.signatures.remove(invalid_signature);
         }
+
+        has_enough_sigs(self, min_group_size)
```

Comment:
Before this, we also need to remove all signatures from nodes that are not among the `sending_nodes`. The `SignedMessage` instance could have been deserialised, not locally constructed, so the check in `add_signature` doesn't guard against this.